File: src/utils/commafy.ts
export default function commafy(input: string | number) {
    return input.toString().replace(/(^|[^\w.])(\d{4,})/g, function (_, $1, $2) {
      return $1 + $2.replace(/\d(?=(?:\d\d\d)+(?!\d))/g, "$&,")
    });
}

File: src/pages/Module/Utils/Commafy.lua
---@class Commafy
--Add commas to a number after being converted to a string
--
--Use `mw.language:formatNum` instead
---@deprecated
---@param str string | number String to add commas to.
---@return string stringWithCommas String with commas.
function commafy(str)
    str = tostring(str)
    return #str % 3 == 0 and str:reverse():gsub("(%d%d%d)", "%1,"):reverse():sub(2) or str:reverse():gsub("(%d%d%d)", "%1,"):reverse()
end

return commafy